app/services/csv/lettings_log_csv_service.rb

```diff
@@ -12,30 +13,126 @@ def to_csv
         csv << @attributes
 
         LettingsLog.all.find_each do |record|
-          csv << @attributes.map do |att|
-            label_from_value(record, att)
-          end
+          csv << @attributes.map { |attribute| get_value(attribute, record) }
         end
       end
     end
 
   private
 
-    def label_from_value(record, att)
-      if %w[la prevloc].include? att
-        label_from_boolean_value(record.send(att))
-      elsif %w[mrcdate startdate voiddate].include? att
-        record.send(att)&.to_formatted_s(:govuk_date)
+    ATTRIBUTES_OF_RELATED_OBJECTS = {
```

Comment:
The convention is at the top, yes.

Reply:
What's the benefit of using this over delegating in the log?

Reply:
the delegating was a little confusing already, and then in particular when we needed to get the codes instead of the human readable values there was an extra layer of weirdness on top because we had to call bla_before_type_cast on some of the values etc.
you can see at the below commit what it had become before moving to this way of doing.
https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/1268/commits/66f5f379fbc8d7fc1618847fbbf5f9c8f23fcb04

in general, it was decided that since there is some fairly case by case mapping between some of these things as they exist in the app and how they should be exported, it would be better to restrict that to this file and make it really explicit if overly verbose, than let it spill into the lettings log class